### PR TITLE
DEVPROD-6928: added tags to Task class

### PR DESCRIPTION
**Pull request by aldo-aguilar**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 3.6.30 - 2024-05-29
```
Comment (zituo-jin) [inline CHANGELOG.md:3]: [nit] I thought this PR was opened on 05/29

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "evergreen.py"
-version = "3.6.29"
+version = "3.6.30"
```
Comment (zituo-jin) [inline pyproject.toml:3]: Can we make this a minor version change according to [semver](https://semver.org/) (I know we've been misusing it)?

**Review by zituo-jin — COMMENTED** [review]
Would you mind giving an example task with `tags` field (just wanted to know how it's used as I don't see it very often on server tasks)?

**Comment by aldo-aguilar** [comment]
@zituo-jin tags aren't on the evergreen UI yet, just in config files and the api. 

[here's a task with tags](https://spruce.mongodb.com/task/mms_e2e_cypress_E2E_CYPRESS_BILLING_a6747c98f656d97008b46baac7e872f6755e9c4b_24_06_17_16_34_48?execution=0&sortBy=STATUS&sortDir=ASC), and[ the api response for the task ](https://evergreen.mongodb.com/rest/v2/tasks/mms_e2e_cypress_E2E_CYPRESS_BILLING_a6747c98f656d97008b46baac7e872f6755e9c4b_24_06_17_16_34_48)

**Review by zituo-jin — COMMENTED** [review]
The change itself LGTM, but I have another question that maybe @rongodb or @gmishkin can help answer. How does foliage access the `tags` field in the task REST endpoint currently (I thought it was via evergreen.py but looks like not)?

**Review by zituo-jin — APPROVED** [review]

**Comment by gmishkin** [comment]
@aldo-aguilar can answer this now! 😁

**Pull request merged**